profiler/controllers/main.py

```diff
@@ -65,6 +67,20 @@ def enable(self):
         core.enabled = True
         ProfilerController.begin_date = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
         ProfilerController.player_state = "profiler_player_enabled"
+
+        # BEGIN CUSTOM CODE
+        profiler_obj = request.env["profiler.profiler"]
+        user = request.env.user
+
+        # Delete user's existing profiler record
+        profiler_id = profiler_obj.search([("user_id", "=", user.id)])
+        profiler_id.unlink()
+
+        # Create new profiler record for the current user
+        profiler_id = profiler_obj.create({"user_id": user.id, "datetime_begin": ProfilerController.begin_date})
+        request.env.cr.commit()
+        # END CUSTOM CODE
```

Comment:
Since this is public fork, let's not worry about `BEGIN`/`END` sections here. We always have a reference back to the original repo and can do proper diffs (compared to what we have to deal with for our normal modules.)